Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ParseTreeValueVisitor.cs

```diff
@@ -87,31 +97,37 @@ private bool ContextHasResult(ParserRuleContext context)
 
         private void Visit(ParserRuleContext parserRuleContext)
         {
-            if (IsUnaryResultContext(parserRuleContext))
-            {
-                VisitUnaryResultContext(parserRuleContext);
-            }
-            else if (parserRuleContext is VBAParser.LExprContext lExpr)
-            {
-                Visit(lExpr);
-            }
-            else if (parserRuleContext is VBAParser.LiteralExprContext litExpr)
-            {
-                Visit(litExpr);
-            }
-            else if (parserRuleContext is VBAParser.CaseClauseContext
-                || parserRuleContext is VBAParser.RangeClauseContext)
-            {
-                VisitImpl(parserRuleContext);
-                StoreVisitResult(parserRuleContext, _inspValueFactory.Create(parserRuleContext.GetText()));
-            }
-            else if (IsBinaryMathContext(parserRuleContext) || IsBinaryLogicalContext(parserRuleContext))
-            {
-                VisitBinaryOpEvaluationContext(parserRuleContext);
-            }
-            else if  (IsUnaryLogicalContext(parserRuleContext) || IsUnaryMathContext(parserRuleContext))
+            switch (parserRuleContext)
             {
-                VisitUnaryOpEvaluationContext(parserRuleContext);
+                case VBAParser.LExprContext lExpr:
+                    Visit(lExpr);
+                    return;
+                case VBAParser.LiteralExprContext litExpr:
+                    Visit(litExpr);
+                    return;
```

Comment:
I believe that you can get the same behavior by calling the `base.Visit()`. Maybe call it in the `default` case where none of the other conditions are met, and remove those trivial cases.

Reply:
The overloaded Visit(...) calls like the one identified above, have logic unique to their VBAParser type (or group of types).  I did some experimenting and have yet to find any way to reuse/leverage the `base.Visit()` (by which I think you mean `IParseTreeVisitor<IParseTreeVisitorResults> Visit(IParserTree)` - if you are referring to different `Visit()` , please let me know).